grpc.go

```diff
@@ -806,6 +807,10 @@ func (a *agentGRPC) SignalProcess(ctx context.Context, req *pb.SignalProcessRequ
 func (a *agentGRPC) WaitProcess(ctx context.Context, req *pb.WaitProcessRequest) (*pb.WaitProcessResponse, error) {
 	proc, ctr, err := a.sandbox.getProcess(req.ContainerId, req.ExecId)
 	if err != nil {
+		// primary process no longer exists so trigger a shutdown of
+		// the gRPC server and thus the agent.
+		a.requestServerStop()
```

Comment:
@sboeuf - `requestServerStop()` seems to *have* to be called here: moving it to `DestroySandbox()` does not dtrt and stops the gRPC server from ending. By having the call here, we're waiting for the shim to call `WaitProcess()`.

What I don't particularly like is the "stop gRPC server on any error from `getProcess()`" approach (similar to https://github.com/kata-containers/runtime/pull/940). I could determine if it was a grpc `NotFound` error I guess?

Reply:
@jodh-intel 
I'm sorry but now that I took a detailed look at the entire code, I think that calling `a.requestServerStop()` from there is not the right thing to do.

`WaitProcess()` is the function that is called by every shim on the host, representing a process. It could be any container process, or any exec'ed process running into an existing container.
Based on this, it does not seem appropriate to ask the server to be stopped on any `WaitProcess()` error. It should be obvious from the code, that we request the server to be stopped when the sandbox is being stopped and removed, or from any piece of code that knows for sure that the agent is done and about to exit.

Based on the current code, I don't understand how a properly running container will end up calling `requestServerStop()` since I don't expect any error to come out of `getProcess()`.

And in case of a multi containers pod, if one container process goes wrong for some reason, do we want to force the grpc server to be stopped, and the agent at the same time?